Core/Ticket System: Ticket comment visual output bug

lucas-wb opened the pull request:
**Changes proposed**:
- Do not send the Gm's name to the ticket handler when commenting on a ticket

**Target branch(es)**: 335 (6.x?)

**Issues addressed**: Closes #16091

**Tests performed**: (Does it build, tested in-game, etc)
Builds, tested commenting on a ticket, assigning tickets, and unassigning tickets

Dire-Tomatoes committed:
Core/Ticket System: Ticket comment visual output bug

Fixes #16091

Commenting on a ticket no longer sends the GM's name to the ticket handler for assignment

Nyeriah commented on src/server/scripts/Commands/cs_ticket.cpp:
```diff
@@ -199,7 +199,7 @@ class ticket_commandscript : public CommandScript
         sTicketMgr->UpdateLastChange();
 
-        std::string msg = ticket->FormatMessageString(*handler, NULL, ticket->GetAssignedToName().c_str(), NULL, NULL, NULL);
+        std::string msg = ticket->FormatMessageString(*handler, NULL, NULL, NULL, NULL, NULL);
```
What's the point of his line after this is removed? It's not passing any data into the function at all, it's all NULLs

Tomatodev replied:
Silly me. Thanks for catching that

Dire-Tomatoes committed:
Update cs_ticket.cpp

Removed useless line entirely

Naios commented on src/server/scripts/Commands/cs_ticket.cpp:
```diff
@@ -199,7 +199,6 @@ class ticket_commandscript : public CommandScript
         sTicketMgr->UpdateLastChange();
 
-        std::string msg = ticket->FormatMessageString(*handler, NULL, ticket->GetAssignedToName().c_str(), NULL, NULL, NULL);
         msg += handler->PGetParseString(LANG_COMMAND_TICKETLISTADDCOMMENT, player ? player->GetName().c_str() : "Console", comment);
```
This will cause an error now...

P-Kito commented on src/server/scripts/Commands/cs_ticket.cpp:
```diff
@@ -199,7 +199,6 @@ class ticket_commandscript : public CommandScript
         sTicketMgr->UpdateLastChange();
 
-        std::string msg = ticket->FormatMessageString(*handler, NULL, NULL, NULL, NULL, NULL);
         msg += handler->PGetParseString(LANG_COMMAND_TICKETLISTADDCOMMENT, player ? player->GetName().c_str() : "Console", comment);
```
```c++
std::string msg = handler->PGetParseString(LANG_COMMAND_TICKETLISTADDCOMMENT, player ? player->GetName().c_str() : "Console", comment);
```

Naios commented:
Could you explain the benefit of this change please?

Dire-Tomatoes committed:
Update cs_ticket.cpp

Sorry this is what I get for not paying attention. msg variable is properly declared now

Tomatodev commented:
The line currently when you comment on an open ticket, you become assigned to it. removing that line will no longer assign you to the ticket when commenting on it.

Naios commented:
`FormatMessageString` and `GetAssignedToName` both never change the assignment of the commented ticket.

Also the current codebase looks like not to change the assignment of the ticket through commenting.
Did you test your change?

Tomatodev commented:
The 3rd argument of `FormatMessageString` is `szAssignedToName` which does this:
`ss << handler.PGetParseString(LANG_COMMAND_TICKETLISTASSIGNEDTO, szAssignedToName);`

The original bug report implies it doesn't actually assign it, but it is only visual. I did test it with this change, and it no longer visually assigns it

Naios commented:
Ah I see, so it's a visual bug only, I guess it says something like `Assigned to` without any name because the empty string is translated to a c-string and the handler thinks it must parse the string.

Maybe you could replace `ticket->GetAssignedToName().c_str()` by `ticket->GetAssignedToName().empty() ? nullptr : ticket->GetAssignedToName().c_str()` which isn't the best style but stillo better then displaying nothing.

Dire-Tomatoes committed:
changed the line per Naois' suggestion, its a case now and won't send nothing
